src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -24,96 +37,145 @@
  *
  * See https://github.com/CrossRef/rest-api-doc
  */
-public class CrossRef {
-    private static final Log LOGGER = LogFactory.getLog(CrossRef.class);
+public class CrossRef implements IdParserFetcher<DOI>, EntryBasedParserFetcher, SearchBasedParserFetcher, IdBasedParserFetcher {
```

Comment:
Instead of the ...Parser...interfaces, I could imagine a syntax similar to 
````java
Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
    return FetcherBuilder.invoke(getURLForID(identifier))
                         .parse(this::getParser)
                         .cleanup(this::doPostCleanup)
                         .fetchSingle();
````
What do you think?

(Anyway, the aim of this PR was to add a single fetcher not to refactor everything)